### PR TITLE
merge edge installer builder and builder role, added kickstart vars

### PR DESCRIPTION
**Pull request by resoluteCoder**

### CODE REVIEW

**Review by maxamillion — APPROVED** [review]

**Review by matoval — APPROVED** [review]

File: roles/build_edge_installer/tasks/main.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
 ---
+- name: Set facts
+  ansible.builtin.set_fact:
+    builder_blueprint_temp: '{{ builder_blueprint_name }}'
```
Comment (maxamillion) [inline roles/build_edge_installer/tasks/main.yml:4]: Why is this necessary?

Comment (resoluteCoder) [reply roles/build_edge_installer/tasks/main.yml:4]: That was the only way I could get the var shared or passed to the builder role. I tried using `defaults_from` but I think its only used when you want to change to a different file within the same role.

Is there another way of doing this? perhaps a cleaner approach? :smile:

Comment (maxamillion) [reply roles/build_edge_installer/tasks/main.yml:4]: commenting here for posterity, we discussed this on Matrix and decided to merge the two roles and have the installer be built based on parameters to the `builder` role ... though we probably want to move that to a different PR away from the kickstart vars

File: roles/builder/tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 ---
+- name: Set fact
+  when: "'-' in builder_compose_type"
+  ansible.builtin.set_fact:
+    a: "{{ builder_compose_type }}"
```
Comment (maxamillion) [inline roles/builder/tasks/main.yml:5]: What's the motivation behind setting this `a` variable? I would suggest we just use `builder_compose_type` everywhere. Thoughts?

File: roles/builder/tasks/main.yml [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
+    builder_compose_type: "{{ builder_compose_type.split('-')[0] + '-commit' if builder_compose_type.split('-')[1] == 'installer' else builder_compose_type }}"
```
Comment (maxamillion) [inline roles/builder/tasks/main.yml:6]: Suggestion: rename this variable to `_image_build_compose_type` and use that in the block below, and keep the original `builder_compose_type` variable as is

**Review by matoval — APPROVED** [review]

**Review by maxamillion — APPROVED** [review]
